.github/workflows/bumpdevversion.yml

```diff
@@ -0,0 +1,120 @@
+      # Step 2a: OLD_VERSION_STRING=6.2.0.dev0+abcdefg,
+      # Step 2b. NEW_VERSION_STRING=6.2.0.dev1+hijklmn
+      - name: Get package's latest git revision and build version string
+        id: version_info
+        run: |
+          ## Step 1a: Get current (old) sha hash from recipe/meta.yaml file (e.g. abcdefghijklmnopqrstuvwxyz12345678901234)
+          export OLD_SHA_HASH=$(yq read recipe/meta.yaml source.git_rev)
+          echo ::set-output name=OLD_SHA_HASH::$OLD_SHA_HASH
+
+          ## Step 1b: Get latest (new) sha hash value from Github (e.g. hijklmnopqrstuvwxyz12345678901234abcdefg)
+          export NEW_SHA_HASH=$(git ls-remote https://github.com/GenericMappingTools/gmt HEAD | cut -f 1)
+          echo ::set-output name=NEW_SHA_HASH::$NEW_SHA_HASH
+
+
+          ## Step 2a. Get current (old) sha hash and version string from recipe/meta.yaml file (e.g. 6.2.0.dev0+abcdefg)
+          export OLD_VERSION_STRING=$(yq read recipe/meta.yaml package.version)
+          echo ::set-output name=OLD_VERSION_STRING::$OLD_VERSION_STRING
+
+          ## Step 2b. Build new version string (e.g. 6.2.0.dev1+hijklmn)
+
+          # Step 2b i. Get current tag of latest dev build (e.g. 6.2.0)
+          export TAG=$(echo $OLD_VERSION_STRING | grep -o -P '.*(?=.dev)')
+
+          # Step 2b ii. Add 1 to the previous dev version number (e.g. 6.2.0.dev0+abcdefg would output 1, i.e. for dev1)
+          export DEV_VERSION_NUM=`expr 1 + $(yq read recipe/meta.yaml package.version | grep -o -P '(?<=dev).*(?=\+)')`
+
+          # Step 2b iii. Get shortened new sha hash value (e.g. hijklmn)
+          export SHORT_HASH=$(echo $NEW_SHA_HASH | cut -c 1-7)
+
+          # Step 3d. Finally build the new version string (e.g. 6.2.0.dev1+hijklmn)
+          echo ::set-output name=NEW_VERSION_STRING::$TAG.dev$DEV_VERSION_NUM+$SHORT_HASH
+
+      # Change development package version and source git revision
+      - name: Update package version and git revision
+        run: |
+          # Update the recipe/meta.yaml file with new version info
+          yq write --inplace recipe/meta.yaml package.version ${{ steps.version_info.outputs.NEW_VERSION_STRING }}
+          yq write --inplace recipe/meta.yaml source.git_rev ${{ steps.version_info.outputs.NEW_SHA_HASH }}
+
+          # Fix incorrectly formatted {{ compiler('c') }}
+          yq write --inplace recipe/meta.yaml requirements.build[1] "{{ compiler('c') }}"
+          sed --in-place "s/'{{ compiler(''c'') }}'/{{ compiler('c') }}/g" recipe/meta.yaml
+
+          # Ensure <two spaces>#<one space>[<expression>] form
+          sed --in-place "s/\ \#\ \[/  # [/g" recipe/meta.yaml
+
+      # Take a look at recipe/meta.yaml contents
+      - name: Check meta.yaml contents
+        run: cat recipe/meta.yaml
+
+      # Generate token from GenericMappingTools bot
+      - uses: tibdex/github-app-token@v1.1.0
+        id: generate-token
+        with:
+          app_id: ${{ secrets.APP_ID }}
```

Comment:
Sorry to bother you @seisman, could you help me to add a secret called **APP_ID**. You'll need to go to https://github.com/GenericMappingTools/gmt-feedstock/settings/secrets and click 'Add secret'

![image](https://user-images.githubusercontent.com/23487320/97063732-e3577180-15fd-11eb-954e-be9e3283e3a8.png)

 The APP_ID should be a short number (5-7 characters) from the https://github.com/apps/genericmappingtools-bot settings page. Forgot to save it when I transferred the bot ownership over at https://github.com/GenericMappingTools/pygmt/pull/646#discussion_r511181579. Thanks!

Reply:
Done.

Reply:
Still some kinks to work out on using Github App token authentication. Currently getting `Error: Resource not accessible by integration`. Will try again next week if I have time.